PhotoScraper/RPI_SIS_PhotoScraper.py

```diff
@@ -115,12 +117,21 @@ def saveImagesToFolder(term, course, class_list):
             if k == "img url":
                 img_url = class_list[i].get(k)
         # download and save the image to a specific folder (term/course_section) from the image url
-        img_name = rcs_id+".png"
-        filepath = path / img_name
         if img_url.split("/")[-1].strip() == "web_transparent.gif":
             print("Skipping {} because no photo on SIS".format(rcs_id))
             continue
         r = requests.get(img_url)
+
+        #Open the image use PIL.Image to deduce the extension
+        img_format = imghdr.what(BytesIO(r.content)).lower()
```

Comment:
You have to have the first argument to `what` be `None` and the second argument be the bytes. The first argument is expected to be a filepath.

Reply:
Aha. You're right. For whatever reason it did work, despite a blob of bytes clearly not being a filepath. I even had the documentation open, that'll teach me not to read. I'll get that fixed.

I find it annoying to have a mix of images that are .jpg and .jpeg when trying to search, and with the exception of .tiff I'm used to seeing mostly 3-character extensions for images. Given the primary purpose of the scraper is for uploading to Submitty, we can scrap the if statement and use .jpeg if you think that's best.

Reply:
Using BytesIO I got: `TypeError: '_io.BytesIO' object is not subscriptable` 

Seems to accept the r.content as the second argument just fine though?

Reply:
Yeah, I didn't read this super closely. `r.content` returns a bytes array which is what we wanted. BytesIO returns a stream over a bytes object (like if we were opening a file). It might be that you could pass an open file handle to the function and it'd work (which is essentially what we're doing BytesIO), but I don't know, and agree in just removing the BytesIO usage and just pass `r.content` directly to `what`.